Source/BidMachineAdapterBannerAd.swift

```diff
@@ -39,11 +39,21 @@ final class BidMachineAdapterBannerAd: BidMachineAdapterAd, PartnerAd {
 
         self.loadCompletion = completion
 
-        // There's no harm in setting the placement ID when loading a bidding ad, but calling
-        // .withPayload(request.adm ?? "") causes an error when BidMachine parses the empty string
-        config.populate { $0.withPlacementId(request.partnerPlacement) }
         if let adm = request.adm {
             config.populate { $0.withPayload(adm) }
+        } else {
+            config.populate {
+                let setting = self.request.partnerSettings["price"]
+                let priceNSNum = setting as? NSNumber
+                guard let price = priceNSNum?.doubleValue else {
+                    let error = error(.loadFailureInvalidAdRequest)
+                    self.log(.loadFailed(error))
+                    completion(.failure(error))
+                    return
+                }
+                $0.withPlacementId(request.partnerPlacement)
+                .appendPriceFloor(price, UUID().uuidString)
```

Comment:
Can you add a comment explaining why a UUID is passed in this method? It's unclear just by reading this code.

Reply:
[Added link to documentation](https://github.com/ChartBoost/chartboost-mediation-ios-adapter-bidmachine/pull/6/commits/ad753efedda7bb874d0c0dee442238889ec6fa72).

Reply:
ty